cmd/lotus-storage-miner/init_restore.go

```diff
@@ -30,250 +31,259 @@ import (
-			}
-		} else {
-			log.Warn("--storage-config NOT SET. NO SECTOR PATHS WILL BE CONFIGURED")
-		}
-
-		log.Info("Restoring metadata backup")
-
-		mds, err := lr.Datastore(context.TODO(), "/metadata")
-		if err != nil {
-			return err
-		}
-
-		bar := pb.New64(st.Size())
-		br := bar.NewProxyReader(f)
-		bar.ShowTimeLeft = true
-		bar.ShowPercent = true
-		bar.ShowSpeed = true
-		bar.Units = pb.U_BYTES
-
-		bar.Start()
-		err = backupds.RestoreInto(br, mds)
-		bar.Finish()
-
-		if err != nil {
-			return xerrors.Errorf("restoring metadata: %w", err)
-		}
+	maddr, err := address.NewFromBytes(abytes)
+	if err != nil {
+		return xerrors.Errorf("parsing actor address: %w", err)
+	}
 
-		log.Info("Checking actor metadata")
+	log.Info("ACTOR ADDRESS: ", maddr.String())
 
-		abytes, err := mds.Get(datastore.NewKey("miner-address"))
-		if err != nil {
-			return xerrors.Errorf("getting actor address from metadata datastore: %w", err)
-		}
+	mi, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+	if err != nil {
+		return xerrors.Errorf("getting miner info: %w", err)
+	}
 
-		maddr, err := address.NewFromBytes(abytes)
-		if err != nil {
-			return xerrors.Errorf("parsing actor address: %w", err)
-		}
+	log.Info("SECTOR SIZE: ", units.BytesSize(float64(mi.SectorSize)))
 
-		log.Info("ACTOR ADDRESS: ", maddr.String())
+	wk, err := api.StateAccountKey(ctx, mi.Worker, types.EmptyTSK)
+	if err != nil {
+		return xerrors.Errorf("resolving worker key: %w", err)
+	}
 
-		mi, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
-		if err != nil {
-			return xerrors.Errorf("getting miner info: %w", err)
-		}
+	has, err := api.WalletHas(ctx, wk)
+	if err != nil {
+		return xerrors.Errorf("checking worker address: %w", err)
+	}
 
-		log.Info("SECTOR SIZE: ", units.BytesSize(float64(mi.SectorSize)))
+	if !has {
+		return xerrors.Errorf("worker address %s for miner actor %s not present in full node wallet", mi.Worker, maddr)
+	}
 
-		wk, err := api.StateAccountKey(ctx, mi.Worker, types.EmptyTSK)
-		if err != nil {
-			return xerrors.Errorf("resolving worker key: %w", err)
-		}
+	log.Info("Initializing libp2p identity")
 
-		has, err := api.WalletHas(ctx, wk)
-		if err != nil {
-			return xerrors.Errorf("checking worker address: %w", err)
-		}
+	p2pSk, err := makeHostKey(lr)
+	if err != nil {
+		return xerrors.Errorf("make host key: %w", err)
+	}
 
-		if !has {
-			return xerrors.Errorf("worker address %s for miner actor %s not present in full node wallet", mi.Worker, maddr)
-		}
+	peerid, err := peer.IDFromPrivateKey(p2pSk)
+	if err != nil {
+		return xerrors.Errorf("peer ID from private key: %w", err)
+	}
 
-		log.Info("Checking proof parameters")
+	return after(api, maddr, peerid, mi)
+}
 
-		if err := paramfetch.GetParams(ctx, build.ParametersJSON(), build.SrsJSON(), uint64(mi.SectorSize)); err != nil {
-			return xerrors.Errorf("fetching proof parameters: %w", err)
-		}
+var restoreCmd = &cli.Command{
```

Comment:
Can we keep this at the top, as it serves as the subcommand definition/entrypoint, and also to minimise the diff noise?